### DISCUSSION TITLE
Visualize Open Geoscience Use Stats & Dependency Relationships - 't21-hack-stats'

### DISCUSSION THREAD

**Discussion by JustinGOSSES** [softwareunderground/transform-2021-hackathon#14] [Projects]
## Description & Goals
Potential idea for Transform21 hackathon project is to visualize engagement and dependency sharing among open-source geoscience code projects cataloged in Awesome-open-geoscience: https://github.com/softwareunderground/awesome-open-geoscience

If this was generated for Awesome-open-geoscience projects, you'd start to get an idea 
- (1) what ones are being reused as dependencies 
- (2) what dependencies are most common among open-source geoscience projects 
- (3) and how has the uptake of different projects changed over time.

## Slack Channel
t21-hack-stats

## People 
Idea by Justin Gosses, unsure if I'll work this one.

## Example Images from NASA repo stats
<img src="https://github.com/JustinGOSSES/llnl.github.io/raw/bcec44e8f80df507d48c507a48889b5607b96af0/assets/images/nasa_starts.png">
<img src="https://github.com/JustinGOSSES/llnl.github.io/raw/nasa/assets/images/nasa_cloud.png">

## Possible Plan
Lawerence Livermore National Laboratory has some nice python code to pull related data from GitHub API and visualize the code repo metadata as a series of d3.js based visualizations in a github pages page https://github.com/LLNL/llnl.github.io

It can pull everything in a list of orgs, users, and/or repos. <- This could be generated via a script that parses https://github.com/softwareunderground/awesome-open-geoscience/blob/master/README.md and pulls out the relevant repos names and adds them to https://github.com/LLNL/llnl.github.io/blob/main/_explore/input_lists.json file. A bash script that can be repeatedly triggered to run in GitHub Actions calls the python scripts and uses a github token to get the public metadata. This is then used to generate the sequence of d3.js based visualizations as seen on:

 https://software.llnl.gov/explore/
 
 
I generated a version that pulls in data from NASA open-source code in a couple hours as seen:
• https://justingosses.github.io/llnl.github.io/explore/
• https://justingosses.github.io/llnl.github.io/explore/dependencies/
• https://justingosses.github.io/llnl.github.io/explore/popular-repos/

You'll still have to change the theme a bit to get rid of all the LLNL branding but doesn't seem too hard. It is basically a jekyll page. There's an MIT license.

## Skills Needed
Someone who is familiar some of jekyll, github API, parsing markdown files, and github actions would be speed things up but required knowledge is not at advanced level.

Non-coders:
- Could use perspectives on whether the visualizations make sense to peopel new to github.
- Could use writing on how to interpret & use the visualizations.

**The end product should gives us some interesting insight into the state of open source subsurface code tools.**

**Comment by RichardScottOZ** [comment]
Sounds like an interesting idea!

**Comment by filippo82** [comment]
[T21 Hackathon Idea Mashup](https://softwareunderground.org/events/2021/4/7/t21-hackathon-idea-mashup) today (or tomorrow depending on your time zone) Wednesday April 14th at 16:00 UTC. Come to Slack to find the link for the meeting!

**Comment by JustinGOSSES** [comment]
Added non-coders needs section

**Comment by JustinGOSSES** [comment]
Project plan is here: https://github.com/softwareunderground/transform-2021-hackathon/projects/2

Repo has been created here: https://github.com/softwareunderground/open_geosciene_code_projects_viz 

Live pages: 
Explore page:  https://softwareunderground.github.io/open_geosciene_code_projects_viz/explore/
Dependencies analysis: https://softwareunderground.github.io/open_geosciene_code_projects_viz/explore/dependencies/
Most popular repositories: https://softwareunderground.github.io/open_geosciene_code_projects_viz/explore/popular-repos/
